src/public/assets/custom/js/album.js

```diff
@@ -0,0 +1,336 @@
+                }
+            });
+        });
+    }
+
+    function mimeIcon(mime) {
+        const defaultIcon = "bi-file-earmark";
+        if (!mime) {
+            return defaultIcon;
+        }
+        const key = mime.split('/')[0];
+        return iconMap.get(key) ?? iconMap.get(mime) ?? defaultIcon;
+    }
+
+    function renderAlbum(album, viewMode) {
+        albumNameElt.innerText = album.name;
+        if (album.files.length === 0) {
+            const emptyWarning = document.createElement("p");
+            emptyWarning.textContent = "Empty Album";
+            albumFilesElt.replaceWith(emptyWarning);
+            return;
+        }
+
+        const newViewMode = document.querySelector("#albumViewControls .btn-outline-primary")?.id;
+        if (newViewMode === "tableView") {
+            resetSelectedFiles("card");
+        } else if (newViewMode === "cardView") {
+            resetSelectedFiles("table");
+        } else {
+            resetSelectedFiles();
+        }
+
+        switch (viewMode) {
+            case "table":
+                renderTable(album);
+                window.location.hash = "table";
+                break;
+            case "card":
+                renderCard(album);
+                window.location.hash = "card";
+        }
+
+
+        function renderTable(album) {
+            site.display(true, albumCardsElt, true);
+            site.display(false, albumFilesElt, true);
+
+            document.getElementById("tableView").classList.replace("btn-outline-primary", "btn-primary");
+            document.getElementById("cardView").classList.replace("btn-primary", "btn-outline-primary");
+
+            if (filesRendered) {
+                return;
+            }
+
+            const tbody = document.createElement("tbody");
+            for (const e of album.files) {
+                const tr = document.createElement("tr");
+                const tdCheckbox = document.createElement("th");
+                tdCheckbox.setAttribute("scope", "row");
+                tdCheckbox.innerHTML = `<input type="checkbox" data-id="${e.id}" id="${e.id}" class="fileCheck" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Select File" />`;
+                tr.appendChild(tdCheckbox);
+
+                const tdName = document.createElement("td");
+                tdName.innerHTML = `<a href="${e.url}" target="_blank">${e.name}</a> ${e.protected ? "<i class='bi bi-file-lock2-fill'></i>" : ""}`;
+                tr.appendChild(tdName);
+
+                const tdSize = document.createElement("td");
+                tdSize.textContent = sizeAsHuman(e.size);
+                tr.appendChild(tdSize);
+
+                const tdActions = document.createElement("td");
+                const directDownload = `<a href="${e.url}?download=true" target="_blank" class="btn btn-outline-primary border-0" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Download"><i class="bi bi-box-arrow-down"></i></a>`;
```

Comment:
^^ should be done using React.js/Vue.js to avoid raw html string in js :)

Reply:
also, vike plugin can be interesting to generate the front-end page, share data context between back/front and use the SSR (it could be a cool improvement)

Reply:
It could be done using react yes. at some point, I shall migrate there. this project was original not really meant to have a complex UI. 

at the same time, i am just kinda happy to use some raw js, keeps the memory fresh of some things some peopke lose with things like react